File: tugasakhir/app/src/main/java/com/example/tugasakhir/BaseActivity.java
package com.example.tugasakhir;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class BaseActivity extends AppCompatActivity {

    public static String nama;
    public static String id;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    public void setLog(String message){
        Log.v("TugasAkhir", message);
    }

    public void setToast(String message){
        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
    }

    public void setIntent(Class<?> activity){
        Intent intent = new Intent(this, activity);
        startActivity(intent);
    }

    public void setNama(String data){
        nama = data;
    }

    public String getNama(){
        return nama;
    }

    public void setId(String data){
        id = data;
    }

    public String getId(){
        return id;
    }
}

File: Jet Pack Compose/okeoce/app/src/main/java/live/hms/myapplication/base/BaseActivity.kt
package live.hms.myapplication.base

import android.content.Intent
import androidx.activity.ComponentActivity
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import android.widget.Toast

open class BaseActivity : ComponentActivity() {

//    public val context = ContextA.current

    open fun setToast (message: String) {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show()
    }

    @Composable
    open fun setText(text: String) {
        Text(text = "$text!")
    }

    fun setActivity(activity: Class<*>) {
        intent = Intent(this, activity)
        startActivity(intent)
        finish()
    }

}